File: src/main/java/org/jdc/template/model/database/main/household/Household.java
/*
 * Household.java
 *
 * Created: 10/14/2013 12:14:38
 */



package org.jdc.template.model.database.main.household;

import android.database.Cursor;

import org.dbtools.android.domain.database.contentvalues.DBToolsContentValues;


public class Household extends HouseholdBaseRecord {


    public Household(Cursor cursor) {
        setContent(cursor);
    }

    public Household(DBToolsContentValues values) {
        setContent(values);
    }

    public Household() {
    }


}

File: app/src/main/kotlin/org/jdc/template/model/domain/Household.kt
package org.jdc.template.model.domain

import kotlinx.datetime.Clock
import org.jdc.template.model.domain.inline.CreatedTime
import org.jdc.template.model.domain.inline.HouseholdId
import org.jdc.template.model.domain.inline.LastModifiedTime
import org.jdc.template.model.domain.inline.LastName
import java.util.UUID

data class Household(
    val id: HouseholdId = HouseholdId(UUID.randomUUID().toString()),
    val name: LastName,

    val created: CreatedTime = CreatedTime(Clock.System.now()),
    val lastModified: LastModifiedTime = LastModifiedTime(Clock.System.now())
)